### ISSUE TITLE
Please include default CSS that works?

### ISSUE THREAD

**Issue by jbardu** [okcoker/taggle.js#80]
I saw the previous issues now closed, and looked at the jsfiddle.
however there are still issues with a default CSS, one that is safe to include within a larger project?

even in the jsfiddle https://jsfiddle.net/okcoker/s5yx2yc1/ - which I only found by looking at the closed issues - there is no default CSS to highlight duplicates - but duplicates are silently refused (enter key feels like it doesn't work).

Also there is no protection from over-running the input div at which point things go wrong, especially with respect to any elements placed after the div.

The docs/examples seem to indicate very simple use but actually those do not work as described .. you do need both a default CSS file and some styling for the DIV etc.

thanks.

**Comment by okcoker** [comment]
Hey @jbardu

> Also there is no protection from over-running the input div at which point things go wrong

What do you mean by this?

I would agree not highlighting a duplicate tag when you are disallowing duplicates is a bit wonky on the UX side, but I'd like to keep the library as lean as possible with the ability to give the developer full control over their styling. Making sure the developer has the appropriate hooks to accomplish what they need to do is important for that reason. That being said, I am not disagreeing with anything you mentioned, so what would you propose? Putting that JSFiddle in the readme? Do you think if that JSFiddle had styling added for highlighting the duplicate tag, it would be more useful?

**Comment by jbardu** [comment]
So in the jsfiddle .. make it smaller as it would be in an input form, also put an element above it.
Now the hint "enter tags" grey text is in the wrong place, because that minimal CSS doesn't have position relative as an attribute.
Then put an element below the div and enter tags until the box fills up.. the element below jumps to the right..

Also if you want to change the spacing and padding of the tags, so they maybe have some margin around them but less padding inside then the hover X is positioned wrongly and hovering makes the length of all tags move which pushes the input caret weirdly onto the line below.

So basically as it stands without being a CSS guru, it's quite hard to integrate even in vanilla form, even though as a package, it's 95% there!

Thanks.

**Comment by jbardu** [comment]
At the risk of annoying you here is a naive modification to the jsfiddle
that I think illustrates the issues I mentioned with integration using the
"minimal" CSS..

https://jsfiddle.net/eb3o47b3/

Enter some tags until the box overflows.
And that's without changing the tag padding, or margin style. If I do
that then the other problem with removal crops up as well.

thanks.

**Comment by okcoker** [comment]
No worries! If a textarea has an absolute height like that, `overflow: scroll` could be a solution. How's this? https://jsfiddle.net/okcoker/eb3o47b3/2/

**Comment by jbardu** [comment]
On Safari at least, that jsfiddle doesn't work at all :(
the scroll bars look terrible and then the tags vanish and scroll up.

I've patched the taggle integration to refuse to add more tags if the total
length of
the tags exceeds X but unfortunately that doesn't stop typing a long tag
before
trying to add it..

Also I'm still stuck with any styling of the tags and the interaction of
that styling
with the "X" to delete option. It seems the default CSS is very carefully
crafted
so that the "X" doesn't change the length of the tags. Any naive CSS
adjustment
breaks the connection, and things start acting strangely.

thanks.

**Comment by okcoker** [comment]
I'm assuming youre on a mac? Scrollbars can be hidden in your system preferences. Otherwise any `overflow: scroll` container will look like you're on windows, which I also agree look terrible. Other than that, you'd have to increase the container height and maybe change `overflow` to `hidden`. This is assuming you want that absolute height. I also noticed you added inline styles which aren't necessary. 

As far as the default styling of the X button, it is a bit more than basic but you can use the devtools to inspect the tags [in the "Misc" section here](https://sean.is/poppin/tags) to view other examples that don't hide the x by default. Removing the `overflow:scroll` and `height` properties will let it grow accordingly. I removed styling for the show/hide functionality and also added a clearfix so the textarea stays around the tags that are added. 

https://jsfiddle.net/okcoker/eb3o47b3/5/

Other than that, I would recommend checking out css articles around the web to get a grasp of how positioning works. [CSS Tricks](https://css-tricks.com/) is a good source.

**Comment by jbardu** [comment]
No argument I can do it all myself.

I just think .. based on your very slick Docs page, it LOOKS like it all
should be very
simple. But actually isn't. So a lot of people like me are seduced into
thinking ok at least basic functionality is plug and play and then they
find out that they've signed up for a lot of integration work.

I can't argue with the price :) but I feel it should either come with a
template CSS that works the same way as the example page without being
glitchy, and without using any style rules that are going to probably exist
in the larger project. Or there should be some section in the docs on
integration challenges. Because the point of using such a widget is to not
have any browser headaches. As of right now I don't really know if what
I've got is glitchy on mobile or opera or IE10 etc.
A default CSS template can have issues filed against it until it does work
ok across all browsers.

**Comment by okcoker** [comment]
All of the SCSS/CSS from the docs page is included in the example folder. Since the rules in an already existing project would most likely interfere with Taggle, it is unrealistic to expect premade styles to work out of the box for every project without bloating all of the existing taggle css rules with every rule that might be in said project. I will not try to attempt that but I think the JSFiddle posted above might be a good start if you want to take a crack at it.

I think the docs could include a section about integration challenges/styling and list the JSFiddle link so I'll add that.
